pylint/checkers/classes/class_checker.py

```diff
@@ -980,7 +980,7 @@ def _check_unused_private_attributes(self, node: nodes.ClassDef) -> None:
                 if attribute.attrname != assign_attr.attrname:
                     continue
 
-                if self._is_type_self_call(attribute.expr):
+                if isinstance(attribute.expr, nodes.Call):
```

Comment:
Resolving the false negative then caused a failure here in another test -- we really just need to test if it's a Call and short-circuit, because `Call` doesn't have a `name` attribute and will crash below.